### PR TITLE
Remove redundant php-docs

### PR DESCRIPTION
**Pull request by malukenho**
I also took the liberty to add some php7 hints to it.

### CODE REVIEW

**Comment by Majkl578** [comment]
Please see https://github.com/doctrine/coding-standard/pull/24 and https://github.com/Majkl578/doctrine2/commit/c2b786d6d3e826b81bb862f5d7a94bf3ba1aa0de (WIP).

File: lib/Doctrine/ORM/Cache/QueryCacheKey.php [hunk 1]
```diff
@@ -32,15 +32,10 @@ class QueryCacheKey extends CacheKey
      */
     public $timestampKey;
 
-    /**
-     * @param string $hash      Result cache id
```
Comment (Majkl578) [inline lib/Doctrine/ORM/Cache/QueryCacheKey.php]: It's ok to remove these, but the variable should be self-explaining. `$hash` is not such case.

File: lib/Doctrine/ORM/Event/OnFlushEventArgs.php [hunk 2]
```diff
@@ -17,19 +17,11 @@ class OnFlushEventArgs extends EventArgs
...
-     */
     public function __construct(EntityManagerInterface $em)
     {
         $this->em = $em;
     }
 
-    /**
-     * Retrieve associated EntityManager.
-     *
-     * @return \Doctrine\ORM\EntityManagerInterface
```
Comment (Majkl578) [inline lib/Doctrine/ORM/Event/OnFlushEventArgs.php]: This is incorrect - EntityManagerInterface becomes mixed.

File: lib/Doctrine/ORM/Event/PostFlushEventArgs.php [hunk 3]
```diff
@@ -17,19 +17,11 @@ class PostFlushEventArgs extends EventArgs
...
-     */
     public function __construct(EntityManagerInterface $em)
     {
         $this->em = $em;
     }
 
-    /**
-     * Retrieves associated EntityManager.
-     *
-     * @return \Doctrine\ORM\EntityManagerInterface
```
Comment (Majkl578) [inline lib/Doctrine/ORM/Event/PostFlushEventArgs.php]: This is incorrect - EntityManagerInterface becomes mixed.

File: lib/Doctrine/ORM/Event/PreFlushEventArgs.php [hunk 4]
```diff
@@ -17,17 +17,11 @@ class PreFlushEventArgs extends EventArgs
...
-    /**
-     * Constructor.
-     */
     public function __construct(EntityManagerInterface $em)
     {
         $this->em = $em;
     }
 
-    /**
-     * @return \Doctrine\ORM\EntityManagerInterface
```
Comment (Majkl578) [inline lib/Doctrine/ORM/Event/PreFlushEventArgs.php]: This is incorrect - EntityManagerInterface becomes mixed.

File: lib/Doctrine/ORM/Mapping/AssociationMetadata.php [hunk 5]
```diff
@@ -51,10 +51,7 @@ class AssociationMetadata implements Property
     /** @var CacheMetadata|null */
     private $cache;
 
-    /**
-     * AssociationMetadata constructor.
-     */
-    public function __construct(string $name)
+     public function __construct(string $name)
```
Comment (Majkl578) [inline lib/Doctrine/ORM/Mapping/AssociationMetadata.php]: Wrong indentation.

File: lib/Doctrine/ORM/Mapping/Driver/Annotation/AnnotationDriver.php [hunk 6]
```diff
@@ -194,7 +194,7 @@ public function getAllClassNames()
                     new \RecursiveDirectoryIterator($path, \FilesystemIterator::SKIP_DOTS),
                     \RecursiveIteratorIterator::LEAVES_ONLY
                 ),
-                '/^.+' . preg_quote($this->fileExtension) . '$/i',
+                '/^.+' . preg_quote($this->fileExtension, '/') . '$/i',
```
Comment (Majkl578) [inline lib/Doctrine/ORM/Mapping/Driver/Annotation/AnnotationDriver.php]: This looks like an unrelated bugfix.

File: lib/Doctrine/ORM/Query/Exec/AbstractSqlExecutor.php [hunk 7]
```diff
@@ -10,6 +10,7 @@
 /**
  * Base class for SQL statement executors.
  * @todo Rename: AbstractSQLExecutor
+ * ???
```
Comment (Majkl578) [inline lib/Doctrine/ORM/Query/Exec/AbstractSqlExecutor.php]: What's wrong here?

File: lib/Doctrine/ORM/Cache/Lock.php [hunk 8]
```diff
@@ -19,20 +16,13 @@ class Lock
...
     {
         $this->value = $value;
         $this->time  = $time ?: time();
     }
 
-    /**
-     * @return \Doctrine\ORM\Cache\Lock
-     */
-    public static function createLockRead()
+    public static function createLockRead(): Lock
```
Comment (Majkl578) [inline lib/Doctrine/ORM/Cache/Lock.php]: Let's not change public API just yet, will be done later everywhere + needs to be documented in UPGRADING.

**Review by Majkl578 — CHANGES_REQUESTED** [review]
Looks ok, but some changes are redundant (either here or in CS 3.0 changes - see comment above).
Actually some of the removed comments here would make sense as part of the CS itself (ForbiddenComments rule), do you mind to send PR? 😉

File: lib/Doctrine/ORM/Query/Exec/AbstractSqlExecutor.php (hunk 7 above)
Comment (malukenho) [reply lib/Doctrine/ORM/Query/Exec/AbstractSqlExecutor.php]: Seems like the class was already renamed. May we can drop the comment?

Comment (Majkl578) [reply lib/Doctrine/ORM/Query/Exec/AbstractSqlExecutor.php]: I don't think it was, the name is same, but casing is different `Sql` vs. `SQL`, hasn't been touched since 2009.

**Comment by Majkl578** [comment]
@malukenho Hi, please rebase onto current master where CS 3.0 is applied (#7057). Then once comments are resolved, it's ready to go. :shipit:

**Comment by malukenho** [comment]
@Majkl578 please, take another look at it.

**Comment by lcobucci** [comment]
@malukenho we have some CS violations on your patch: https://travis-ci.org/doctrine/doctrine2/jobs/340480444#L554

Can you please check those?

**Comment by carusogabriel** [comment]
@malukenho This is being doing in #7046 :)

File: lib/Doctrine/ORM/Query/AST/Functions/ConcatFunction.php [hunk 9]
```diff
@@ -38,7 +38,7 @@ public function getSql(SqlWalker $sqlWalker)
             $args[] = $sqlWalker->walkStringPrimary($expression);
         }
 
-        return call_user_func_array([$platform, 'getConcatExpression'], $args);
+        return $platform->getConcatExpression(...$args);
```
Comment (carusogabriel) [inline lib/Doctrine/ORM/Query/AST/Functions/ConcatFunction.php]: @Majkl578 @Ocramius Maybe something for our CS as well?

Comment (Ocramius) [reply lib/Doctrine/ORM/Query/AST/Functions/ConcatFunction.php]: @carusogabriel not really that important, plus `call_user_func_array()` is quite magic, so I'd avoid fiddling with it at CS level.

**Review by Ocramius — APPROVED** [review]
LGTM :ship:
